mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java

```diff
@@ -273,6 +276,14 @@ private static void checkResiduals(CombinedScanTask task) {
     });
   }
 
+  private FileIO fileIO(Table table) {
+    if (table.io() instanceof HadoopFileIO) {
+      return new HadoopFileIO(((HadoopFileIO) table.io()).conf());
```

Comment:
Why does this create a new `HadoopFileIO`? The table's `FileIO` should work fine.

I think this might be based on the [handling in Spark](https://github.com/apache/incubator-iceberg/blob/master/spark/src/main/java/org/apache/iceberg/spark/source/IcebergSource.java#L239-L247), which creates a new one that wraps a broadcasted manifest. But there is no broadcast for MR so I don't think we need this method. It should work fine to use `table.io()`.